instrumentation/opentelemetry-instrumentation-psycopg2/tests/test_psycopg2_integration.py

```diff
@@ -63,6 +64,17 @@ def cursor(self):
     def get_dsn_parameters(self):  # pylint: disable=no-self-use
         return {"dbname": "test"}
 
+    def __setattr__(self, name, value):
```

Comment:
Instead of doing this I think adding something like `self.assertTrue(foo._is_instrument_by_opentelemetry is True)` would be a lot more explicit

Reply:
Thanks for your reply. How about I add "self.assertFalse(hasattr(cnx, "_is_instrumented_by_opentelemetry"))" in test case. It will be failed with existing code in main branch, but pass with the changes. If this makes sense to you, I will update the code and push.

Reply:
Yeah, that would work too.

Reply:
Thank you for your reply, I will make the changes and push shortly.

Reply:
I guess we don't have tests right now to test for duplicate instrumentation. I think since we are changing this to depend on the existence of `ObjectProxy` we should test that at least (since we are removing the reliance on `_is_instrumented_by_opentelemetry` )

Reply:
@xrmx @lzchen I updated PR, could you please help to review and suggest if there is anything I need to change? Thanks.